src/main/java/com/googleintern/wfm/ruleengine/model/UserModel.java

```diff
@@ -50,4 +52,13 @@ public static Builder builder() {
 
     public abstract UserModel build();
   }
+
+  public boolean isConflictUserPair(UserModel comparedUser) {
+    return workforceId() == comparedUser.workforceId()
+        && workgroupId() == comparedUser.workgroupId()
+        && skillIds().containsAll(comparedUser.skillIds())
+        && roleIds().containsAll(comparedUser.roleIds())
+        && roleSkillIds().containsAll(comparedUser.roleSkillIds())
+        && !poolAssignments().containsAll(comparedUser.poolAssignments());
```

Comment:
equals()?

Reply:
The skill ids, role ids, role skill ids and pool assignments are not necessary to be equal if two users are conflict. For example:
User 1: skill ids={111, 222}, role ids={333}, permission set ids={5656, 2020}
User 2: skill ids={111, 222}, role ids={}, permission set ids ={5656, 2020, 2048}
All types of skill ids and role ids for User 2 are covered by User 1, but the permission part for User 2 are not covered.